repl.fnl

```diff
@@ -0,0 +1,45 @@
+(local coroutine (require :coroutine))
+(local fennel (require :fennel))
+(local jeejah (require :jeejah))
+(local view (require :fennelview))
+
+(fn fennel-middleware
+  [f msg]
+  (match msg.op
+    "load-file" (let [f (assert (io.open msg.filename "rb"))]
+                  (tset msg
+                        :op "eval"
+                        :code (-> f
+                                  (: :read "*all")
+                                  (: :gsub "^#![^\n]*\n" "")))
+                  (: f :close))
+    (f msg)))
+
+(local repl-coro-freq 0.05)
+
+(fn run
+  [server]
+  (let [repl-coro server
+        repl-spin (fn [] (coroutine.resume repl-coro))
+        repl-chk (fn [] (not= (coroutine.status repl-coro) "dead"))]
+    (hs.timer.doWhile repl-chk repl-spin repl-coro-freq)))
+
+(fn start-opts
+  [opts]
+  (let [server (jeejah.start nil opts)]
+    server))
+
+(fn start
```

Comment:
The start method should support options.

I would recommend adding the following require:
```fennel
(local {:merge merge} (require :lib.functional))
```

Then changing the definition of start to:

```fennel

(local default-opts 
  {:fennel     true 
   :middleware fennel-middleware
   :serialize  hs.inspect})


(fn start
  [custom-opts]
  (let [ server (jeejah.start (. custom-opts :port) (merge {} default-opts custom-opts))]
    server))
```

This will allow users to customize the repl to their needs but will also work out of the box

Reply:
Done. I had to do a bit different to handle `custom-opts` not being provided.